lua/roslyn/init.lua

```diff
@@ -86,6 +86,10 @@ local function run_roslyn(exe, target, config)
         }
     end
 
+    if vim.fn.executable(exe) == 1 then
```

Comment:
what I am worried about here is if this function just checks for the +x bit, and the dll counts as being executable to this method. Would you be able to test this on your end @seblj ?

Reply:
`vim.fn.executable('/home/mnj/.local/share/nvim/roslyn/Microsoft.CodeAnalysis.LanguageServer.dll')` returns `0`.
The DLL does not have executable permission bit set.

Reply:
Hmm, I am actually thinking about just doing a breaking change, and having the user having to pass a table of `{ "dotnet", "some dll file here" }`, if they want a custom path. And then also just omit `dotnet` if it is not needed🤔 I think this would allow to cleanup a bit regarding the mason stuff. What do you think?

Reply:
It's maybe unfortunate, but maybe we should just remove the entire check for if the file exists or not then. If it doesn't exist, vim.system will fail with no such file or directory.

It would be a worse message for the user to get, but it just painful to deal with running with dotnet, regular binary or mason (if it gets merged hopefully), so maybe just let it error if the user haven't installed it correctly or give the wrong path.

Ideas?

Reply:
Hm, too be honest, I'm not really sure. My lua experience is slim to none, so I'd say whichever you think would be easier to maintain. On the other hand, as a general rule of thumb, simple _and_ flexible is probably the easiest approach? ie the `{ "dotnet", "some dll file here" }` idea

Reply:
Okay, let's do the breaking change. A couple of things then:

- In the check for if the server is installed or not, I think we can revert it back to how it was, and then add a `config.exe == nil` check, so it becomes something like:

`if config.exe == nil and not vim.uv.fs_stat(exe[2]) and not vim.uv.fs_stat(mason_installation)`

So that we can check the default case, but if the user passes the `exe` option, it is more their responsibility to do it correct.

- We need to change a little bit here, as we already then have a table. Something simple would be to just do a `vim.list_extend()` with the extra args for the server. Just make sure that we clone the table first, since lis_extend mutates the `dest` argument. I think something like `vim.deep_clone` or something exist. Otherwise just a naive solution is also okay

- We also need to just update the types from string to string[] a couple of places for the exe option

Reply:
hey, finally got the time to have a stab at this, haven't tested it yet but feel free to leave any feedback in the meantime